### PR TITLE
Update MANIFEST.in

### PR DESCRIPTION
**Pull request by dhirschfeld**
Ensure cython files are packaged in the sdist

### CODE REVIEW

**Comment by dhirschfeld** [comment]
The `cython` files are missing from the sdist:

image

**Comment by dhirschfeld** [comment]
Error looks unrelated - a hash mismatch with `vcpython27` download:
```
vcpython27 v9.0.0.30729 [Approved]
vcpython27 package files install completed. Performing other installation steps.
Downloading vcpython27 
  from 'https://download.microsoft.com/download/7/9/6/796EF2E4-801B-4FC4-AB28-B59FBF6D907B/VCForPython27.msi'

Download of VCForPython27.msi (83.82 MB) completed.
Error - hashes do not match. Actual value was '15C51E531251F617F7E216975A84581D'.
ERROR: Checksum for 'C:\Users\VssAdministrator\AppData\Local\Temp\chocolatey\vcpython27\9.0.0.30729\VCForPython27.msi' did not meet '4e6342923a8153a94d44ff7307fcdd1f' for checksum type 'md5'. Consider passing the actual checksums through with --checksum --checksum64 once you validate the checksums are appropriate. A less secure option is to pass --ignore-checksums if necessary.
The install of vcpython27 was NOT successful.
Error while running 'C:\ProgramData\chocolatey\lib\vcpython27\tools\chocolateyinstall.ps1'.
```

**Comment by jborean93** [comment]
Thanks for the PR, I've rerun the tests to actually see what the sdist contains when CI builds it. Unfortunately it still doesn't have the Cython files

image

This is the same process that builds the sdist when a release is published so the lack of files in a CI run would also mean the release won't have them.

Also just a side question, is there a reason why the wheels won't work on Windows for you? Just trying to get an understanding of the user case.

**Comment by dhirschfeld** [comment]
We use `conda` because it's got a proper dependency resolver. If you pip install a package into an existing analytics environment you're playing Russian roulette with your results.

Sure, `pip` is getting it's own dependency resolver but the entire scientific python stack already has that with `conda`. The scientific python stack is ~90% linking to compiled dependencies which is exactly the problem `conda` was designed to solve and consequently it's a technically superior solution than wheels. Statically compiling dependencies into every binary is very memory/size inefficient and also very complex when having to avoid symbol clashes from duplicated dependencies.

Correctly specifying the dependencies and constraints to get the scientific python stack to work well together is also a hugely complex endeavour but the scientific python community (in the form of conda-forge) is there to help and make sure things work. When wheels don't work there aren't the people there to fix them because the scientific python community is invested in `conda` IME.

So, to summarise, `conda` can do everything `pip` can, it does it better, and it can also do a lot that `pip` can't - I also manage my R environment with `conda` and also use it for most of my dev tools too.

* http://technicaldiscovery.blogspot.com/2013/12/why-i-promote-conda.html
* https://uwekorn.com/2019/09/15/how-we-build-apache-arrows-manylinux-wheels.html

***Disclaimer:*** These are my own, (highly opinionated) opinions!

**Comment by dhirschfeld** [comment]
I'm popping up here because we use [`dask`](https://dask.org/), a distributed parallel processing framework, which relies on [`fsspec`](https://filesystem-spec.readthedocs.io/en/latest/), which has made `smbprotocol` a dependency - https://github.com/intake/filesystem_spec/pull/340

**Comment by dhirschfeld** [comment]
> it does it better,

Though, TBF it builds on top of a lot of the functionality provided by pip/setuptools (for Python), but it does a lot more - it's a system/language agnostic package manager.

**Comment by dhirschfeld** [comment]
I'm no expert on `MANIFEST.in` but I thought that would've worked - will look into that 🤔

**Comment by dhirschfeld** [comment]
`recursive-include` seems to do the job in my local testing on Windows

**Comment by jborean93** [comment]
Looks good, just checked the sdist built in CI and it contains the .pyx and .pyd files.

Thanks for the explanation, conda is not something I've used before and I've just dealt with pip and wheels for better or for worse :) One future thing I'm hoping to implement is to include the .c files that cython generates from the .pyx/.pyd files in the sdist so end users don't need cython to actually create the compiled extensions. That wouldn't be an issue for your use case right?

**Comment by dhirschfeld** [comment]
Not for our use case. When I did a simple `python setup.py sdist` I did see the `.c` files included locally but I guess the CI is doing something different.

**Comment by jborean93** [comment]
Currently CI just creates the sdist based on what is in the git repo which at the build time is cleared and doesn't contain the cython generated .c files. If you were to build it locally after you've generated the .c files that would make sense but the artifact sent to PyPI will not contain them (yet). At some point in the future I'm wanting to generate them and include it in the sdist but for now that's not the case.

Ultimately what it means is that to install pyspnego from the sdist with the SSPI extensions present you will still need cython to be installed at install/build time. I was just merely checking that this new feature in the future won't break how you currently use the library.

**Comment by dhirschfeld** [comment]
@jborean93 - I was meaning to ask if you'd like to be added as a maintainer of the conda-forge recipes?
* https://github.com/conda-forge/smbprotocol-feedstock
* https://github.com/conda-forge/python-gssapi-feedstock
* https://github.com/conda-forge/pyspnego-feedstock

This gives you merge rights on the repos but it also means you'll get some notifications, mostly for new PRs. Don't feel you have to at all - as a user of both `conda` and your packages I'm happy to keep the recipes up-to-date in collaboration with the rest of the community.

If you don't want to now, at any time you can put in a PR to add your name to the maintainers which will then automatically give you merge rights - e.g.
https://github.com/conda-forge/python-gssapi-feedstock/blob/cf211b012867e566534a2982f2310beba80b2470/recipe/meta.yaml#L47-L48

**Comment by jborean93** [comment]
Thanks, I appreciate the offer but I probably don't have time to really add on some more maintenance, especially since I don't fully understand/use the conda landscape. I do appreciate you working on them though and making these libraries easier for others to use and for opening the PR here.

I've release [v0.1.1](https://github.com/jborean93/pyspnego/releases/tag/v0.1.1) to [PyPI](https://pypi.org/project/pyspnego/) which includes your fixes here. I've checked the sdist there and like the one produced in CI it includes the cython files in `_sspi_raw`.
